plugin/pluginCore.js

```diff
@@ -29,37 +28,32 @@ exports.runPa11y = async function({ htmlFilePaths, testMode, debugMode }) {
 exports.generateFilePaths = async function({
   fileAndDirPaths, // array, mix of html and directories
   PUBLISH_DIR,
+  build,
   testMode,
   debugMode
 }) {
-  let htmlFilePaths = [];
-  for (fileAndDirPath of fileAndDirPaths) {
-    const fullDirPath = path.join(PUBLISH_DIR, fileAndDirPath);
-    if (fs.statSync(fullDirPath).isDirectory()) {
-      let subPaths = await walk(fullDirPath);
-      htmlFilePaths = htmlFilePaths.concat(subPaths);
-    } else {
-      htmlFilePaths.push(fullDirPath);
-    }
-  }
-  return htmlFilePaths;
+  const htmlFilePaths = await Promise.all(
+    fileAndDirPaths.map(fileAndDirPath => findHtmlFiles(`${PUBLISH_DIR}/${fileAndDirPath}`, build))
+  )
+  return [].concat(...htmlFilePaths)
 };
 
-var walk = async function(dir, filelist) {
-  var files = await readDir(dir);
-  filelist = filelist || [];
-  await Promise.all(
-    files.map(async function(file) {
-      const dirfile = path.join(dir, file);
-      if (fs.statSync(dirfile).isDirectory()) {
-        filelist = await walk(dirfile + '/', filelist);
-      } else {
-        if (dirfile.endsWith('.html')) filelist.push(dirfile);
-      }
-    })
-  );
-  return filelist;
-};
+const findHtmlFiles = async function(fileAndDirPath, build) {
+  if (await isDirectory(fileAndDirPath)) {
+    const fileInfos = await readdirp.promise(fileAndDirPath, { fileFilter: '*.html' })
+    return fileInfos.map(({ fullPath }) => fullPath)
+  }
+
+  if (!(await isFile(fileAndDirPath))) {
+    build.failBuild(`Input "checkPaths" does not exist: ${fileAndDirPath}`)
```

Comment:
wait.. why fail build? should we just warn instead?

also i think the wording can be better.. maybe `Warning: Folder ${fileAndDirPath} was provided in "checkPaths", but does not exist - it either indicates something went wrong with your build, or you can simply delete this folder from your "checkPaths" in netlify.toml`

Reply:
Fixed!